apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/ExplanationOfBenefitResourceProvider.java

```diff
@@ -347,10 +357,13 @@ public Bundle findByPatient(
 
     eobs.sort(ExplanationOfBenefitResourceProvider::compareByClaimIdThenClaimType);
 
+    Bundle bundle =
+        TransformerUtils.createBundle(paging, eobs, loadedFilterManager.getTransactionTime());
+
     // Add bene_id to MDC logs
-    TransformerUtils.logBeneIdToMdc(beneficiaryId);
+    LoggingUtils.logBenesToMdc(bundle);
 
-    return TransformerUtils.createBundle(paging, eobs, loadedFilterManager.getTransactionTime());
+    return bundle;
```

Comment:
My fault for leading you down this path but this piece of code can be reverted so that we will correctly log the bene_id when it has no EOBs (at the small risk of logging fake bene_ids).